### ISSUE TITLE
FinalLocalVariable to work with lombok

### ISSUE THREAD

**Issue by bbonanno** [checkstyle/checkstyle#7038]
When using lombok there is simpler way to write final variables, so instead of 
```java
final String foo = "bar";
```

We can write

```java
val foo = "bar";
```

Both expressions will compile to the same code, so `FinalLocalVariable` should not flag the "lombok way" as an error as it is a `final` variable.

**Comment by strkkk** [comment]
@bbonanno checkstyle does not analyze bytecode, only sources

**Comment by rnveach** [comment]
Duplicate of https://github.com/checkstyle/checkstyle/issues/5470

**Comment by bbonanno** [comment]
@strkkk I'm not talking about bytecode, I'm just saying that the regex or whatever analyses the code should contemplate that syntax as valid.
Anyway, as @rnveach pointed out, there was a similar issue (that for some reason I didn't see) in which is state that support for lombok will not be added, so I'll close this issue.

Thanks
